src/ims/model/_convert.py

```diff
@@ -35,6 +35,11 @@ def freezeStrings(strings: Iterable[str]) -> frozenset[str]:
     return frozenset(strings)
 
 
+def dedupeStrings(strings: Iterable[str]) -> tuple[str, ...]:
+    deduped = {s: None for s in strings}
```

Comment:
return tuple(frozenset(strings))

Reply:
The problem is that frozenset causes the loss of order. Dictionaries remove duplicate keys, but maintain insertion order

```
>>> print(tuple(frozenset(["a", "b", "c"])))
('b', 'a', 'c')
```

Reply:
huh